circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/S3S3Copier.java

```diff
@@ -158,8 +159,8 @@ private void initialiseAllCopyRequests() {
         .info("Initialising all copy jobs");
     AmazonS3URI sourceBase = toAmazonS3URI(sourceBaseLocation.toUri());
     AmazonS3URI targetBase = toAmazonS3URI(replicaLocation.toUri());
-    srcClient = s3ClientFactory.newSourceInstance(sourceBase, s3s3CopierOptions);
-    targetClient = s3ClientFactory.newTargetInstance(targetBase, s3s3CopierOptions);
+    srcClient = s3ClientFactory.newInstance(sourceBase, s3s3CopierOptions);
```

Comment:
but now you get the same client with the assume role for both source and target that's incorrect right?
I thought you need "normal" client for source and "assume-role" client for target.

Reply:
we've established it does make much of a difference, since the role has read access to the source